### ISSUE TITLE
Entry size limits and multivalue attribute limits

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#312]
In order to prevent some classes of denial of service, we should consider entry size limits to prevent excess growth of entries that accounts can self write to, especially where there are multivalue attributes that can be created without limit.

**Comment by MinhPhan8803** [comment]
Is this still being considered

**Comment by Firstyear** [comment]
Yes it is, I'm probably looking at it soon. It's not as simple as sticking in some limits though as they likely need to come from schema with default values.

**Comment by MinhPhan8803** [comment]
I see. What are some specific reasons they need to come from schema?

**Comment by Firstyear** [comment]
If a user develops their own custom attributes, then our limits may not be correct - perhaps the user wants a text field that has 1000 characters rather than 200? So an admin has to be able to lift and alter those limits from schema when their define the attribute type. Similar you can have a limit on the number attrs in a multivalue attribute but that may not always be appropriate. For example we could limit phone numbers to 4, but then there are other cases where you may want more phone numbers. So that's why - because when people extend schema and add their own attributes they way want *alternate* limits than we provide.

**Comment by MinhPhan8803** [comment]
Thanks for clarifying :)
